### ISSUE TITLE
Machine-readable yotta ls output

### ISSUE THREAD

**Issue by bremoran** [ARMmbed/yotta#186]
Some common operations would be easier if `yotta ls` produced machine-readable output.  In particular, a way of removing the tree diagram and colour would help.

For example an output like the following would provide easy parsing
```
$ yotta ls --no-format
module#version
dependency1#version
linked-dependency -> link-target#version
```

**Comment by autopulated** [comment]
`--plain`?

**Comment by bremoran** [comment]
--porcelain?
See: [git-status](http://git-scm.com/docs/git-status) --porcelain

**Comment by hugovincent** [comment]
I literally shuddered just now trying `--porcelain?` as output reminded me of SVN...

**Comment by danros** [comment]
This is something we will need also, for all yotta commands really.

**Comment by autopulated** [comment]
So, some commands (like build) stream the output from subprocess as they run, and add their own output before/after that.

I'd like to add a `--json` type flag, which processes command output into a very robust machine readable form, but unfortunately that is not easily compatible with streaming output.

So here's what I propose.

`yotta` accepts `--plain` as a top level argument (before any subcommands). This will prevent coloured output in all commands, and modify the output of commands like `yotta list` to remove other fancy formatting.

If that proves insufficient, we can add additional `--json` output flags for specific commands where it makes sense.

In the future --plain might be the default if yotta detects stdout is a pipe, too.

**Comment by bremoran** [comment]
That sounds good to me.

**Comment by autopulated** [comment]
See #226

**Comment by autopulated** [comment]
This was fixed since yotta 0.13.0, when the `yt ls --json` format was stabilised.
